### PR TITLE
Feature/save

### PR DESCRIPTION
**Pull request by ajthinking**
* Basic save by polling every 5 seconds
* Data saved into `Project->workbench_data` 

Improvements we can do:
* Request only when something is changed. Means we need to dispatch a `save()` in every action though?
* Keep the poll, but rinse the `workbench_data` to only send keys that have changed

### CODE REVIEW

File: src/PipeDream.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
-import defaultStore from './defaultStore/index.js';
+import defaultStore from './defaultStore/index.js'
+const mergeJSON = require('deepmerge')
 
 export default class PipeDream {
     constructor(options) {
-        this.options = options
+        this.options = mergeJSON(this.defaultOptions(), options)
```
Comment (adhikasp) [inline src/PipeDream.js]: This is nice! Doesn't cross my mind to use deepmerge here.

File: src/defaultStore/state.js [hunk 2]
```diff
@@ -1,9 +1,15 @@
-export default function(options) {
-
-    if (typeof window.__ENV__ === 'undefined') {
-        window.__ENV__ = 'dummy';
+function getDefault(key, defaultValue) {    
+    if (typeof window.__ENV__.workbench_data[key] !== 'undefined') {
```
Comment (adhikasp) [inline src/defaultStore/state.js]: Cool 🌟 
Why not combine the null and undefined check in 1 statement?

File: src/components/AppWorkspace.vue [hunk 3]
```diff
@@ -15,6 +15,31 @@
...
+                    this.saveDebounce = setTimeout(() => func.apply(context, args), delay)
+                }.bind(this)
+            }
+
+            this.$store.subscribe((mutation, state) => {
+                // Assume each mutation produces a new state - no need for hash
+                debounce(async () => {
+                    let result = await this.$store.dispatch('save')
+                    console.log(result)
+                }, 3500)()
```
Comment (adhikasp) [inline src/components/AppWorkspace.vue]: Move this magic number to options.
And I guess it saves every 3.5 sec instead of 5? 🙈

Comment (ajthinking) [reply src/components/AppWorkspace.vue]: Good idea! On the composer package we can afford high update frequency, on pipedream.ai lower

File: src/defaultStore/state.js (hunk 2 above)
Comment (ajthinking) [reply src/defaultStore/state.js]: Yes think this whole thing can be done better

File: src/PipeDream.js (hunk 1 above)
Comment (ajthinking) [reply src/PipeDream.js]: 👍

File: src/components/AppWorkspace.vue (hunk 3 above)
Comment (ajthinking) [reply src/components/AppWorkspace.vue]: Fixed

**Pull request merged**